src/components/InputProblem.tsx

```diff
@@ -1,7 +1,92 @@
+import React, { useState, useCallback } from 'react'
 import { BaseProblemProps } from './ProblemSetBlock'
+import { Formik, Form, Field, ErrorMessage } from 'formik'
+import { mathifyElement } from '../lib/math'
+import * as Yup from 'yup'
 
-interface InputProblemProps extends BaseProblemProps {}
+interface InputProblemProps extends BaseProblemProps {
+  comparator: string
+}
+
+interface InputSchema {
+  response: string | number
+}
+
+interface InputFormValues {
+  response: string
+}
+
+export const InputProblem = ({
+  solvedCallback, exhaustedCallback, allowedRetryCallback,
+  solution, retryLimit, content, comparator, encourageResponse, buttonText, correctResponse
+}: InputProblemProps): JSX.Element => {
+  const [retriesAllowed, setRetriesAllowed] = useState(0)
+  const [inputDisabled, setInputDisabled] = useState(false)
+  const [feedback, setFeedback] = useState('')
+
+  const schema = (): Yup.SchemaOf<InputSchema> => {
+    if (comparator === 'integer') {
+      return Yup.object({
+        response: Yup.number().integer().typeError('Please provide an Integer').required('Please provide an Integer')
+      })
+    }
+    if (comparator === 'float') {
+      return Yup.object({
+        response: Yup.number().typeError('Please provide an number').required('Please provide a number')
+      })
+    }
+    return Yup.object({
+      response: Yup.string().trim().required('Please provide valid input')
+    })
+  }
+
+  const contentRefCallback = useCallback((node: HTMLDivElement | null): void => {
+    if (node != null) {
+      mathifyElement(node)
+    }
+  }, [])
+
+  const handleSubmit = async (values: InputFormValues): Promise<void> => {
+    if (values.response.trim() === solution.trim()) {
```

Comment:
I don't think this comparison is sufficient, and we probably need one that is a function of the actual `comparator`. For example, if I define the problem as:

```html
<div class="os-raise-ib-pset-problem" data-problem-type="input" data-solution="42.0" data-problem-comparator="float">
```

I should be able to provide `42` or `42.0` or `42.0000` and get it correct. 

I would suggest:

* For `text` use `===` after converting both to lower case (e.g. so `red` matches a solution definition of `Red`)
* For `int` compare using results from `parseInt` 
* For `float` compare using results from `parseFloat`